src/views/waypoint/home/index.tsx

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-case-declarations */
+/* eslint-disable react/no-array-index-key */
+import React, { ReactElement } from 'react'
+import DocsLayout from 'layouts/docs'
+import GetStarted, { GetStartedProps } from './components/get-started'
+import Cards, { CardProps, CardInterface } from './components/cards'
+import Heading, { HeadingProps } from './components/heading'
+
+function WaypointLanding({
+  CONTENT,
+}: {
+  CONTENT: { heading: string; subheading: string; blocks: Block[] }
+}): ReactElement {
+  return (
+    <div>
+      <h1>{CONTENT.heading}</h1>
+      <p>{CONTENT.subheading}</p>
+      <LandingBlocks blocks={CONTENT.blocks} />
+    </div>
+  )
+}
+
+interface BlockBase {
+  type: 'heading' | 'get_started' | 'cards'
+  // TODO: properties below are already defined,
+  // TODO: eg in HeadingProps, GetStartedProps, & CardProps,
+  // TODO: but still getting TypeScript compile errors in switch
+  // TODO: blocks below.
+  // TODO: Need to figure out how to do this type union (?) thing properly
+  heading?: string
+  __heading_slug?: string
+  level?: number
+  iconSvg?: string
+  text?: string
+  link?: {
+    url: string
+    text: string
+  }
+  columns?: number
+  cards?: CardInterface[]
+}
+
+type Block =
+  | (HeadingProps & BlockBase)
+  | (GetStartedProps & BlockBase)
+  | (CardProps & BlockBase)
```

Comment:
I've been trying to get this "mixed type" card to work, but can't seem to avoid Typescript errors without the duplicative properties declared in the `BlockBase` type. Going to keep looking into this and try to sort it out 👍

CC @BRKalow @ashleemboyer in case either of y'all have some TypeScript wizardry or pattern suggestions to spare.

Reply:
Have updated this to avoid the duplicative optional properties, but have had to case `block` as specific types in an `if else` block. This feels a lot better, but I suspect it isn't the truly correct way of doing things. Will keep digging 👀